Tests/Api/CurrentWeatherDataResultTest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Savchenko\Bundle\OpenWeatherMapBundle\Tests\Api;
+
+use GuzzleHttp\Psr7\Response;
+use Savchenko\Bundle\OpenWeatherMapBundle\Api\CurrentWeatherDataResult;
+
+class CurrentWeatherDataResultTest extends \PHPUnit_Framework_TestCase
+{
+    use FakeWeatherDataTrait;
+
+    private $response;
+
+    public function setUp()
+    {
+
+        $this->response = $this->prophesize(Response::class);
+    }
+
+    /**
+     * @test
+     */
+    public function It_should_save_the_response_when_constructed_from_response()
+    {
+
+        $this->response->getBody()->willReturn($this->getWeatherData());
```

Comment:
You mean mocking the response? Yeah, stupid since they are immutable. Thanks for your valuable feedback.